tests/test_agglomerate.cc

```diff
@@ -0,0 +1,263 @@
+/*************************************************************************
+ * Copyright (c) 2017 by the mfmg authors                                *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE agglomerate
+
+#include "main.cc"
+
+#include <mfmg/amge.hpp>
+
+#include <deal.II/distributed/tria.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+#include <array>
+#include <boost/mpi.hpp>
+
+template <int dim>
+std::vector<unsigned int> test(boost::mpi::communicator world)
+{
+  dealii::parallel::distributed::Triangulation<dim> triangulation(world);
+  dealii::FE_Q<dim> fe(1);
+  dealii::DoFHandler<dim> dof_handler(triangulation);
+
+  dealii::GridGenerator::hyper_cube(triangulation);
+  triangulation.refine_global(3);
+  dof_handler.distribute_dofs(fe);
+
+  mfmg::AMGe<dim, dealii::TrilinosWrappers::MPI::Vector> amge(world,
+                                                              dof_handler);
+
+  std::array<unsigned int, dim> agglomerate_dim;
+  for (unsigned int i = 0; i < dim; ++i)
+    agglomerate_dim[i] = i + 2;
+  amge.build_agglomerates(agglomerate_dim);
+
+  std::vector<unsigned int> agglomerates;
```

Comment:
Just do `agglomerates(dof_handler.n_active_cells())` and then use `[]` so that you don't do array resize too often.